src/iothread.c

```diff
@@ -100,10 +100,10 @@ void fetchClientFromIOThread(client *c) {
     }
     /* Unbind connection of client from io thread event loop. */
     connUnbindEventLoop(c->conn);
-    freeClientDeferredObjects(c, 1); /* Free deferred objects. */
     /* Now main thread can process it. */
     c->running_tid = IOTHREAD_MAIN_THREAD_ID;
     resumeIOThread(c->tid);
+    freeClientDeferredObjects(c, 1); /* Free deferred objects. */
```

Comment:
Is it `fetching`, not `unblocking`?
Is this modification to avoid pausing io thread for too long?

Reply:
Pausing io thread means it is blocked
yes